plugins/mcreator-core/blockly/js/mcreator_blocks.js

```diff
@@ -800,5 +808,14 @@ Blockly.defineBlocksWithJsonArray([
         ],
         "output": "Direction",
         "colour": "20"
+    },
+    {
+        "type": "damagesource_constant",

```

Comment:
This could be defined in JSON. I'd change the text of this procedure to "Damage source of type: (SELECTOR)" and the id to something like `damagesource_from_type`, since you could create a DamageSource with a source entity and a immediate source entity.

Reply:
Then, we could keep the `damagesource_constant` procedure, but rename it to `damagetype_constant`, so the `damagesource_from_type` procedure block can either take this `damagetype_constant` block, or an entity block

Reply:
The issue is that this would not work for 1.19.2